### DISCUSSION TITLE
Why is ASCII DEL 0x7F excluded?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#121] [Q&A] [answered]
In Forth 200x Draft 19.1 why is ASCII DEL 0x7F excluded from section 3.1.2.2 Control Characters?  DEL is part of ASCII and treated as a control character, typically `^?` on Unix-ish systems.

**Comment by ruv** [comment, marked as answer]
I don't think 0x7F is excluded. It is just not mentioned explicitly, but it's implicitly included in "all non-graphic characters":

> All _non-graphic characters_ included in the **implementation-defined character set** are defined in this standard as **control characters**. _In particular_, the characters {0 . . . 31}, which could be included in the implementation-defined character set, are control characters.

(emphasis added by me)

The characters sub-set {0 . . . 31} is mentioned as an example.

**Reply by SirWumpus** [reply]
This draft never fails to amaze in what it won't commit to.  3.1.2.2 is the first time `implementation-defined character set` is mentioned (out of 6 references in the document), yet its not discussed before that nor any rationale.  They state ASCII is used, of which 0..31 are well defined controls including DEL, so how / when / why would you have an `implementation-defined character set` - maybe some rationale for 3.1.2, 3.1.2.1, 3.1.2.2 might help, for example:

       An `implementation-defined character set` can be any character set where 7-bit ASCII is a proper subset.
       For example UTF-8 encoding, Extended ASCII, Latin1 to name a few.

**Reply by ruv** [reply]
You could submit a [proposal](https://forth-standard.org/proposals) of such an improvement.

I would only say "_shall_ be a superset of" instead of "_can_ be" (see [Chapter 2](https://forth-standard.org/standard/notation#chapter.2)). And it should be appended into [3.1.2](https://forth-standard.org/standard/usage#subsection.3.1.2) (not [3.1.2.2](https://forth-standard.org/standard/usage#subsubsection.3.1.2.2)).
